### ISSUE TITLE
Can this program work on 128bit range?

### ISSUE THREAD

**Issue by aaelick** [JeanLucPons/Kangaroo#122]
Or it can handle absolute maximum 125-bit as stated in readme? Thanks.

**Comment by Baskentliia** [comment]
This kangaroo program is limited to 125 Bits. So 130 Bits cannot be used to solve the puzzle!!! This program needs to be updated. We are waiting for the new version from the person who made this program, I hope he sees us.

**Comment by aaelick** [comment]
> This kangaroo program is limited to 125 Bits. So 130 Bits cannot be used to solve the puzzle!!! This program needs to be updated. We are waiting for the new version from the person who made this program, I hope he sees us.

My question is not for 129-bits for #130 puzzle, but for 128-bit. Why it can't work with 128-bit range? Where in the code 128-bit variables uses 3-bits for storing additional data, not the value itself? Thanks.

**Comment by Baskentliia** [comment]
The program is limited to 125 bit width, that is, the maximum distance between the start and end range is 125 bits.
128 bit won't work
127 bit won't work
126 bit won't work
It works at 125 bits.

Recommendation: You should divide Puzzle 130 into many intervals and bring it to 125 bit level.

We are waiting for an update from the person who created this program.

**Comment by sari777** [comment]
> > This kangaroo program is limited to 125 Bits. So 130 Bits cannot be used to solve the puzzle!!! This program needs to be updated. We are waiting for the new version from the person who made this program, I hope he sees us.
> 
> My question is not for 129-bits for #130 puzzle, but for 128-bit. Why it can't work with 128-bit range? Where in the code 128-bit variables uses 3-bits for storing additional data, not the value itself? Thanks.

this program requires public key.
128 bits address has no leak public key.

**Comment by sssergy** [comment]
> > > This kangaroo program is limited to 125 Bits. So 130 Bits cannot be used to solve the puzzle!!! This program needs to be updated. We are waiting for the new version from the person who made this program, I hope he sees us.
> > 
> > 
> > My question is not for 129-bits for #130 puzzle, but for 128-bit. Why it can't work with 128-bit range? Where in the code 128-bit variables uses 3-bits for storing additional data, not the value itself? Thanks.
> 
> this program requires public key. 128 bits address has no leak public key.


A 130-bit key can be safely reduced to 128 bits.

**Comment by sari777** [comment]
> > > > This kangaroo program is limited to 125 Bits. So 130 Bits cannot be used to solve the puzzle!!! This program needs to be updated. We are waiting for the new version from the person who made this program, I hope he sees us.
> > > 
> > > 
> > > My question is not for 129-bits for #130 puzzle, but for 128-bit. Why it can't work with 128-bit range? Where in the code 128-bit variables uses 3-bits for storing additional data, not the value itself? Thanks.
> > 
> > 
> > this program requires public key. 128 bits address has no leak public key.
> 
> A 130-bit key can be safely reduced to 128 bits.

yes. it is only possible if you already know the first 2-3 hex digits of the private key.(in case of 130bit)
for example "320000000000000000000000000000000:32fffffffffffffffffffffffffffffff"
kangaroo is designed for dp collisions attacks. not for random attacks.
for random attacks, we can try keyhunt of alberto.

**Comment by sssergy** [comment]
> > > > > This kangaroo program is limited to 125 Bits. So 130 Bits cannot be used to solve the puzzle!!! This program needs to be updated. We are waiting for the new version from the person who made this program, I hope he sees us.
> > > > 
> > > > 
> > > > My question is not for 129-bits for #130 puzzle, but for 128-bit. Why it can't work with 128-bit range? Where in the code 128-bit variables uses 3-bits for storing additional data, not the value itself? Thanks.
> > > 
> > > 
> > > this program requires public key. 128 bits address has no leak public key.
> > 
> > 
> > A 130-bit key can be safely reduced to 128 bits.
> 
> yes. it is only possible if you already know the first 2-3 hex digits of the private key.(in case of 130bit) for example "320000000000000000000000000000000:32fffffffffffffffffffffffffffffff" kangaroo is designed for dp collisions attacks. not for random attacks. for random attacks, we can try keyhunt of alberto.

200000000000000000000000000000000 You can subtract immediately, then the key will be either 129 or even smaller, down to 125 bits.
But, of course, we will not be able to know the exact range)

**Comment by GiladLeef** [comment]
For puzzle 130 see https://puzzlesearch.github.io/

**Comment by ejrhe** [comment]
> This kangaroo program is limited to 125 Bits. So 130 Bits cannot be used to solve the puzzle!!! This program needs to be updated. We are waiting for the new version from the person who made this program, I hope he sees us.

![Image](https://github.com/user-attachments/assets/5a3e2de6-73d2-4eb6-9bce-4d66db16fdde) then how is capable of running on full 256-bit key space?
